src/main/java/org/spongepowered/common/mixin/core/advancement/MixinAdvancementList.java

```diff
@@ -58,8 +58,8 @@
 
     @Shadow @Final private static Logger LOGGER;
 
-    @Shadow @Final private Map<ResourceLocation, Advancement> advancements;
-    @Shadow @Final private Set<Advancement> roots;
+    @Shadow @Final private final Map<ResourceLocation, Advancement> advancements = new AdvancementMap();
```

Comment:
Double check what the AP output is when compiling this mixin, I'm not sure whether `@Mutable` would be required. Pinging @Aaron1011 @Mumfrey for clarifying input?